src/libsyntax/ast.rs

```diff
@@ -1354,6 +1355,8 @@ pub enum StrStyle {
 #[derive(Clone, RustcEncodable, RustcDecodable, Debug, Hash, PartialEq)]
 pub struct Lit {
     pub node: LitKind,
+    pub token: token::Lit,
```

Comment:
Let's maybe add a comment here explaining why we need both `LitKind` and `token::Lit` and that we eventually plan to phase out `LitKind`. The problem with such a big incremental refactoring is that, once you are half-way through it, it's easy to forget which way you were going :D Explaining the intention in the comment should help.

I've seen a cute hack at JetBrains for this: when `X` is going to be replaced with `Y` across codebase, first `Y` is added alongside `X`. Then, `X` is marked as deprecated, and all the call-sites are marked with `#[allow(deprecated)]` (IDE helps a lot here). At this stage, new usages of `X` are not added, because they trigger deprecation warning, but old usages continue to work, without causing warnings. Then old usages are incrementally removed.

Reply:
This depends on how much I manage to refactor today, perhaps I'll remove `node` entirely, then the deprecation will be unnecessary, otherwise I'll add the relevant comments.